From d1b86069c9ffa2747642cf163fd65eef0ae7f39c Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Thu, 12 Sep 2024 10:33:00 +0200
Subject: [PATCH 1112/1695] dts: stm32mp: add reset support for remoteproc on
 stm32mp2

Remoteproc expose the RESET related to M33 to non secure world with SCMI.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git d25ade808cfca4b0fa833334e9c27a4d509128c1 dts: stm32mp: add reset support for remoteproc on stm32mp2]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I6676cb7d414a833854df89901c85d04ff862172f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/403990
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 1 +
 core/arch/arm/dts/stm32mp231.dtsi | 1 +
 core/arch/arm/dts/stm32mp251.dtsi | 1 +
 3 files changed, 3 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index b58d24d97..748c74d02 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -125,6 +125,7 @@
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
+			#reset-cells = <1>;
 			reg = <0 0>;
 			resets = <&rcc C2_R>,
 				 <&rcc C2_HOLDBOOT_R>;
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index ea63798a3..a03958d0b 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -126,6 +126,7 @@
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
+			#reset-cells = <1>;
 			reg = <0 0>;
 			resets = <&rcc C2_R>,
 				 <&rcc C2_HOLDBOOT_R>;
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index c3415cb7e..3b78050a6 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -126,6 +126,7 @@
 
 		m33_rproc: m33@0 {
 			compatible = "st,stm32mp2-m33";
+			#reset-cells = <1>;
 			reg = <0 0>;
 			resets = <&rcc C2_R>,
 				 <&rcc C2_HOLDBOOT_R>;
-- 
2.47.2

